src/PayzenGatewayFactory.php

```diff
@@ -13,17 +15,28 @@
  */
 class PayzenGatewayFactory extends GatewayFactory
 {
+    private $transactionId;
+
+    public function __construct(
+        array $defaultConfig = array(),
+        GatewayFactoryInterface $coreGatewayFactory = null,
+        TransactionIdInterface $transactionId = null
+    ) {
+        parent::__construct($defaultConfig, $coreGatewayFactory);
+        $this->transactionId = $transactionId ?? new IdGeneratedByFile($defaultConfig['directory'] ?? sys_get_temp_dir());
+    }
+
     /**
      * Builds a new factory.
      *
-     * @param array                        $defaultConfig
+     * @param array $defaultConfig
      * @param GatewayFactoryInterface|null $coreGatewayFactory
-     *
+     * @param TransactionIdInterface|null $transactionId
      * @return PayzenGatewayFactory
      */
-    public static function build(array $defaultConfig, GatewayFactoryInterface $coreGatewayFactory = null): PayzenGatewayFactory
+    public static function build(array $defaultConfig, GatewayFactoryInterface $coreGatewayFactory = null, TransactionIdInterface  $transactionId = null): PayzenGatewayFactory
```

Comment:
This is not compatible with Payum/Core:
[PayumBuilder::buildAddedGatewayFactories()](https://github.com/Payum/Core/blob/1cb0e0eab4fa939328c2e1620ce7eb71fe1be366/PayumBuilder.php#L533C16-L533C83) won't give a value to $transationId argument.
I think default translation id generator should be defined in the default config.